src/sensai/evaluation/eval_stats.py

```diff
@@ -0,0 +1,122 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar, List, Union, Dict
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib import pyplot as plt
+
+from ..models.base import FitterModel, PredictorModel
+
+TEvalStats = TypeVar("TEvalStats", bound='EvalStats')
+TMetric = TypeVar("TMetric", bound="Metric")
+
+
+# TODO or not TODO: the inheritance structure here creates a circular dependency of eval stats and metrics.
```

Comment:
Out of the many TODOs that I have added, I think this one deserves particular attention. I was really sad when I realized that I could not separate metrics from eval stats, it blew up the corresponding classification and regression modules. What do you think?